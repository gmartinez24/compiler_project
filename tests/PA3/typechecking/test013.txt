// Spencer Banasik
// Ensure that variables are not coerced
main 

int x;

{
    x = 10 && true;

    call printInt(x);
    call println();
    if (x == 1) then
        call printInt(x); // Should not occur
    else
        x = 5;
        call println(x);
    fi; 
}.