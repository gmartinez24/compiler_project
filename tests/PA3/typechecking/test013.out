Error type-checking file.
TypeError(8,13)[Cannot compute int and bool.]
TypeError(8,5)[Cannot assign ErrorType(Cannot comput int and bool.) to int.]